File: SmartMed/backend/models/Caregiver.js
const mongoose = require('mongoose');

const caregiverSchema = new mongoose.Schema({
    userId: { type: String, required: true, index: true },
    name: { type: String, required: true },
    email: { type: String },
    phone: { type: String },
    relationship: { type: String, default: 'Family' },
    active: { type: Boolean, default: true },
    createdAt: { type: Date, default: Date.now }
});

module.exports = mongoose.model('Caregiver', caregiverSchema);

File: SmartMed/android/app/src/main/java/com/smartmed/app/data/model/Caregiver.java
package com.smartmed.app.data.model;

import com.google.gson.annotations.SerializedName;

/**
 * Caregiver model.
 */
public class Caregiver {

    @SerializedName("_id")
    private String id;

    @SerializedName("userId")
    private String userId;

    @SerializedName("name")
    private String name;

    @SerializedName("email")
    private String email;

    @SerializedName("phone")
    private String phone;

    @SerializedName("relationship")
    private String relationship;

    @SerializedName("active")
    private boolean active;

    public Caregiver() { this.active = true; }

    public String getId() { return id; }
    public void setId(String id) { this.id = id; }

    public String getUserId() { return userId; }
    public void setUserId(String userId) { this.userId = userId; }

    public String getName() { return name; }
    public void setName(String name) { this.name = name; }

    public String getEmail() { return email; }
    public void setEmail(String email) { this.email = email; }

    public String getPhone() { return phone; }
    public void setPhone(String phone) { this.phone = phone; }

    public String getRelationship() { return relationship; }
    public void setRelationship(String relationship) { this.relationship = relationship; }

    public boolean isActive() { return active; }
    public void setActive(boolean active) { this.active = active; }
}
